File: ilia/src/models/product.ts
import { z } from "zod";

export type Product = z.infer<typeof ProductSchema>;

export type Rating = {
	rate: number;
	count: number;
};

export const ProductSchema = z.object({
	id: z.number(),
	title: z.string(),
	price: z.number(),
	image: z.string(),
	description: z.string(),
	category: z.string(),
	rating: z.object({
		rate: z.number(),
		count: z.number(),
	}),
});

File: ilia-api/Services/Entities/Product.cs
namespace Services.Entities
{
    public class Product : BaseEntity
    {
        public string Title { get; set; }
        public float Price { get; set; }
        public string Image { get; set; }
        public string Description { get; set; }
        public float Rating { get; set; }
        public List<Order> Orders { get; set; }
    }
}